### PR TITLE
[WIP] Implementation of theme support for Rocket.Chat.

### PR DESCRIPTION
**Pull request by noor-binte-amir**
#### Changes:
* Provides a theme fragment for switching themes.
* Left toggles for Dark Theme and right toggles for Black Theme.
* All code changes are done in the thememanager package.
* Project resources have been changed.
* A themelibrary with its own resources has been added.
* Some colors are yet to be adjusted.

#### Screenshots:
Screenshot_20190618-223829
Screenshot_20190618-223820
Screenshot_20190618-223755

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,22 +18,20 @@
         android:networkSecurityConfig="@xml/network_security_config"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
+        android:theme="@style/AppTheme"
         tools:replace="android:name">
-
+        <!--android:theme="@style/BlackTheme" Placed above-->
```
Comment (philipbrito) [inline app/src/main/AndroidManifest.xml:23]: Do we need these comments on this file? If not, make sure to remove then later.

File: app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt [hunk 2]
```diff
@@ -57,7 +57,10 @@ class MessageViewHolder(
             button_join_video_call.setOnClickListener { joinVideoCallListener(it) }
 
             image_avatar.setImageURI(data.avatar)
-            text_content.setTextColor(if (data.isTemporary) Color.GRAY else Color.BLACK)
+            if (data.isTemporary){
+                text_content.setTextColor(Color.GRAY)
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt:61]: This way the text color will be always gray. What about the data is not temporary?

File: app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt [hunk 3]
```diff
@@ -1032,7 +1031,7 @@ class ChatRoomPresenter @Inject constructor(
                         ChatRoom(
                             id = id,
                             subscriptionId = subscriptionId,
-                            parentId = parentId,
+                            //parentId = parentId,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt:1034]: ? It won't compile. Why commenting this needed parameter?

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt [hunk 4]
```diff
@@ -274,14 +276,14 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
     private fun setupListeners() {
         text_server_name.setOnClickListener {
             ServersBottomSheetFragment().show(
-                activity?.supportFragmentManager,
+                activity!!.supportFragmentManager,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt:279]: Avoid `!!`. It was better when it was declared as the way it was before (`activity?.supportFragmentManager`)

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt [hunk 5]
```diff
@@ -274,14 +276,14 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
...
                 chat.rocket.android.servers.ui.TAG
             )
         }
 
         text_sort_by.setOnClickListener {
             SortingAndGroupingBottomSheetFragment().show(
-                activity?.supportFragmentManager,
+                activity!!.supportFragmentManager,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt:286]: Avoid `!!`. It was better when it was declared as the way it was before (`activity?.supportFragmentManager`)

File: app/src/main/java/chat/rocket/android/db/DatabaseManager.kt [hunk 6]
```diff
@@ -401,13 +401,13 @@ class DatabaseManager(val context: Application, val serverUrl: String, val token
...
     }
 
-    private fun mapLastMessageText(message: LastMessage?): String? = message?.let { lastMessage ->
-        if (lastMessage.message?.isEmpty() == true && lastMessage.attachments?.isNotEmpty() == true) {
-            context.getString(R.string.msg_sent_attachment)
-        } else {
-            lastMessage.message
-        }
-    }
+//    private fun mapLastMessageText(message: LastMessage?): String? = message?.let { lastMessage ->
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/db/DatabaseManager.kt:404]: ? 
Why is this required function commented?

**Review by philipbrito — CHANGES_REQUESTED** [review]
@NBAMIR I had to stop reviewing it since I believe you need to track the things on your repository with the main stream because right now I can see lot of stuffs that are on the main stream and not here making it hard to review. Please update your branch - there is lots of things not related to what this PR is supposed to add here.

**Comment by NBAMIR** [comment]
> @NBAMIR I had to stop reviewing it since I believe you need to track the things on your repository with the main stream because right now I can see lot of stuffs that are on the main stream and not here making it hard to review. Please update your branch - there is lots of things not related to what this PR is supposed to add here.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (noor-binte-amir) [reply app/src/main/AndroidManifest.xml:23]: Oh no we don't. These are just for me and I will be removing them later.

File: app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt (hunk 2 above)
Comment (noor-binte-amir) [reply app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt:61]: If it isnt temporary, it uses the color defined in the xml. However, I have to change this to use colors other than gray depending on the theme.

File: app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt (hunk 3 above)
Comment (noor-binte-amir) [reply app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt:1034]: I was receiving errors for the parentID reference. Peter was as well while he was using JDK 11 but using JDK 10 fixed it for him. I followed the Rocket.Chat readme instructing us to use Java 8 and compiled my project with jdk 1.8.0_152 accordingly. I'm still seeing the errors though so this was a temporary fix for me. I don't intend on pushing this change to Rocket.Chat repository. If you have any idea why I am getting an error on this reference, do let me know.

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt (hunk 4 above)
Comment (noor-binte-amir) [reply app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt:279]: Same as above, I was getting errors on this line and its a temporary fix.

File: app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt (hunk 5 above)
Comment (noor-binte-amir) [reply app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt:286]: Same as above, I was getting errors on this line and its a temporary fix.

File: app/src/main/java/chat/rocket/android/db/DatabaseManager.kt (hunk 6 above)
Comment (noor-binte-amir) [reply app/src/main/java/chat/rocket/android/db/DatabaseManager.kt:404]: Same as above.

**Comment by noor-binte-amir** [comment]
> @NBAMIR I had to stop reviewing it since I believe you need to track the things on your repository with the main stream because right now I can see lot of stuffs that are on the main stream and not here making it hard to review. Please update your branch - there is lots of things not related to what this PR is supposed to add here.

The PR I made for the Rocket.Chat repository only pushed my changes but seems like this PR is showing changes I fetched as well. Let me see how I can reference specific commits only and make a new PR. Apologies for the inconvenience!
